firestore.rules

```diff
@@ -294,7 +294,7 @@ service cloud.firestore {
         // teachers can only delete their own documents
         allow delete: if isAuthedTeacher() && userIsResourceUser();
         // teachers can read their own documents or other documents in their network
-        allow read: if isAuthedTeacher() && (userInResourceTeachers() || resourceInTeacherNetworks());
+        allow read: if  (isAuthed() &&  resource == null) || (isAuthedTeacher() && (userInResourceTeachers() || resourceInTeacherNetworks()));
```

Comment:
do we want any auth'd user? or some subset (e.g. students & teachers?)

Reply:
>do we want any auth'd user? or some subset (e.g. students & teachers?)

Wouldn't something like the following be sufficient without opening things up quite as far?

```suggestion
        allow read: if  (isAuthed() && classInResourceClasses() && resource == null) || (isAuthedTeacher() && (userInResourceTeachers() || resourceInTeacherNetworks()));
```

Reply:
I'm not sure! I see `classInResourceClasses` checks `request.auth.token.class_hash in resource.data.classes;`  My understanding is that if the document doesn't exist that will return false because resource.data.classes wouldn't exist? But I can try to play around with the tests and see.

>

Reply:
Now that you mention it, that seems right, so for now just stick with what you had.